### PR TITLE
Tempo: Support multiple filter expressions for service graph queries

### PR DESCRIPTION
**Pull request by domasx2**
**What is this feature?**

This PR adds ability pro provide multiple promql filter expressions when making a service map query with Tempo data source. In this case promql OR query is used, result is a union of results for each fitler expression. 

`target.serviceMapQuery` now accepts array of strings in addition to single string. It is fully backwards compat.

Only frontend  Tempo data soruce is upgraded, not adding support for multiple filter expressions to service graph query editor at this time. 

**Why do we need this feature?**

App o11y has a need to render service graph that contains all nodes matching given filter criteria + their immediate siblings, and have correct request rates for the edges. 

**Who is this feature for?**

Currently this is needed for app o11y plugin.

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by javiruiz01 — APPROVED** [review]
LGTM 🚀

File: public/app/plugins/datasource/tempo/datasource.test.ts [hunk 1]
```diff
@@ -562,19 +574,114 @@ describe('Tempo service graph view', () => {
...
+          datasourceUid: 'prom',
+        },
+      },
+    });
+    const promMock = prometheusMock();
+    setDataSourceSrv(dataSourceSrvWithPrometheus(promMock));
+    const response = await lastValueFrom(
+      ds.query({
+        targets: [
+          { queryType: 'serviceMap', serviceMapQuery: ['{ foo="bar" }', '{baz="bad"}'], refId: 'foo', filters: [] },
```
[the remark this answers is not in the document]
Comment (domasx2) [reply public/app/plugins/datasource/tempo/datasource.test.ts:637]: :D I'm pretty sure it doesnt matter, white space here is has no significance in promql

File: public/app/plugins/datasource/tempo/datasource.ts [hunk 2]
```diff
@@ -1256,24 +1262,33 @@ function getServiceGraphView(
...
-    serviceMapQuery = serviceMapQueryMatch[1];
+  const serviceMapQueries = Array.isArray(serviceMapQuery) ? serviceMapQuery : [serviceMapQuery];
+  const metricParamsArray = serviceMapQueries.map((query) => {
+    // remove surrounding curly braces from serviceMapQuery
+    const serviceMapQueryMatch = query.match(/^{(.*)}$/);
+    if (serviceMapQueryMatch?.length) {
+      query = serviceMapQueryMatch[1];
+    }
+    // map serviceGraph metric tags to serviceGraphView metric tags
+    query = query.replace('client', 'service').replace('server', 'service');
```
[the remark this answers is not in the document]
Comment (domasx2) [reply public/app/plugins/datasource/tempo/datasource.ts:1278]: yeah, I thought this was weird :sweat_smile:

File: public/app/plugins/datasource/tempo/ServiceGraphSection.tsx [hunk 3]
```diff
@@ -67,7 +67,9 @@ export function ServiceGraphSection({
     );
   }
 
-  const filters = queryToFilter(query.serviceMapQuery || '');
+  const filters = queryToFilter(
+    (Array.isArray(query.serviceMapQuery) ? query.serviceMapQuery[0] : query.serviceMapQuery) || ''
```
[the remark this answers is not in the document]
Comment (domasx2) [reply public/app/plugins/datasource/tempo/ServiceGraphSection.tsx:71]: App o11y does not use Tempo service map query editor. We're going for an opinionated experience where queries are created programatically based on context and some high level filters, and are not exposed directly to users.

A specific problem is that  customers want to filter app o11y service map by environment (or other attrs), and they want to see nodes that match selected environment + nodes they are directly talking to. So we can't filter this with a single promql filter query, we need to filter `serviceMapQuery=['{server_deployment_environment="$env}',  '{client_deployment_environment="$env"}']` to get a union of both.

It also enables us to target specifc service and it's sibilings only, by filtering `serviceMapQuery=['{server="$name"}', '{client="$name"}']`

Related app o11y PR: https://github.com/grafana/app-observability-plugin/pull/767

image

**Review by aocenas — APPROVED** [review]
I think this makes sense. I guess at some point we should probably have similar UI in the tempo query editor

File: public/app/plugins/datasource/tempo/ServiceGraphSection.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (joey-grafana) [reply public/app/plugins/datasource/tempo/ServiceGraphSection.tsx:71]: Nice, makes sense and thanks for the example 👍

**Review by joey-grafana — APPROVED** [review]

**Pull request merged**